Sources/NIOConcurrencyHelpers/lock.swift

```diff
@@ -29,17 +29,21 @@ import Glibc
 /// `SRWLOCK` type.
 public final class Lock {
 #if os(Windows)
-    fileprivate let mutex: UnsafeMutablePointer<SRWLOCK> =
+    private let mutex: UnsafeMutablePointer<SRWLOCK> =
         UnsafeMutablePointer.allocate(capacity: 1)
+#elseif canImport(Darwin)
+    private let mutex: os_unfair_lock_t = os_unfair_lock_t.allocate(capacity: 1)
 #else
-    fileprivate let mutex: UnsafeMutablePointer<pthread_mutex_t> =
+    private let mutex: UnsafeMutablePointer<pthread_mutex_t> =
         UnsafeMutablePointer.allocate(capacity: 1)
 #endif
 
     /// Create a new lock.
     public init() {
 #if os(Windows)
         InitializeSRWLock(self.mutex)
+#elseif canImport(Darwin)
+        self.mutex.pointee = .init()
```

Comment:
Strictly we're supposed to initialise an `os_unfair_lock` to `OS_UNFAIR_LOCK_INIT`. Can you add a comment that indicates that this is equivalent?